[11.0][ADD] mass_mailing_newsletter_welcome_mail

cristinamartinrod opened the pull request:
This module extends the functionality of the website popup newsletter subscription form sending to new subscribers an automatic welcome email and allow you to customize it.

cc @Tecnativa

Commit:
[ADD][11.0] mass_mailing_newsletter_welcome_mail

yajo commented:
Is this ready to review?

pedrobaeza commented on mass_mailing_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,38 @@
+                <div>Dear <b>${object.name}</b>,
+                    <br/>
+                    <br/>
+                    <h3>Thanks for signing up four our newsletter!</h3>
```
s/four/for

cristinamartinrod replied:
Done

pedrobaeza commented on mass_mailing_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,38 @@
+                    <br/>
+                    <h3>Thanks for signing up four our newsletter!</h3>
+                    <p>
+                        Enjoy a discount code for 20% off your first order
```
This shouldn't be included here in this mail, but customize the template on the customer.

cristinamartinrod replied:
This is part of the default template of the newsletter in Odoo, something very common, that's why I included it here. I delete it.

pedrobaeza commented on mass_mailing_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,38 @@
+            <field name="trigger">on_create</field>
+            <field name="active" eval="True"/>
+            <field name="template_id" ref="email_template"/>
+            <field name="filter_domain">[["list_ids","=","Newsletter"]]</field>
```
Why this filter? By default, it should be in any mass mailing list, and be customized on customer.

cristinamartinrod replied:
The same that the previous default text, the subscription is usually to the newsletter and to set a filter for default, easier to modify for the user. I delete it

pedrobaeza replied:
But this doesn't count with translations for example. If there's already a list in data with an XML-ID, use `eval` and `ref` for setting this field with the ID of this mass mailing list by default. That's something reasonable, but not to make the comparison by name.

cristinamartinrod replied:
Yes, just was a little help/guide for users, but it's ok, I removed it to avoid doubts.

pedrobaeza replied:
But my question remains unanswered: is there a mailing list defined in data with that name? If so, definitively it's advisable to set it as said, due to Odoo intention is to have that one as newsletter. If not, then put all of them.

cristinamartinrod replied:
Yes, it's a default list set on data in mass_mailing

pedrobaeza replied:
OK, then it can be interesting to be added, but I let you to decide. If we add it, then:

- Change module title to reflect this: s/mailing list/newsletter.
- Should the module be renamed to `mass_mailing_newsletter_welcome_mail`?
- Put `<field name="filter_domain" eval="[['list_ids', '=', ref('xml_id_of_newsletter')]]"/>`

If not added:

- Change module description to say that is for all mailing list, as right now references the popup only.

cristinamartinrod replied:
I think it's useful if it's configured by default a specific list, so I changed name, descriptions and settings for it.

cristinamartinrod commented:
Now is ready for you review @Yajo :)

yajo commented on mass_mailing_newsletter_welcome_mail/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+              'Odoo Community Association (OCA)',
+    'category': 'Social',
+    'version': '11.0.1.0.0',
+    'website': 'https://github.com/OCA/website',
```
```suggestion
    'website': 'https://github.com/OCA/social',
```

cristinamartinrod replied:
Done

yajo commented on mass_mailing_newsletter_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,32 @@
+            <field name="model_id" ref="mass_mailing.model_mail_mass_mailing_contact"/>
+            <field name="use_default_to" eval="True"/>
+            <field name="lang">${object.env.context.get("lang")}</field>
+            <field name="subject">Welcome to the mailing list</field>
```
```suggestion
            <field name="subject">Welcome to the mailing list ${object.list_id.name}</field>
```

cristinamartinrod replied:
We changed this yesterday, customizing only the Newsletter list. That's the reason we re-named the module. I understand that this is a good contribution for any list, even if it's only for Newsletter, it's the same.

yajo commented on mass_mailing_newsletter_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,32 @@
+            <field name="body_html" type="html">
+                <div>Dear <b>${object.name}</b>,
+                    <br/>
+                    <h3>Thanks for signing up for our newsletter!</h3>
```
```suggestion
                    <h3>Thanks for signing up for ${object.list_id.name}!</h3>
```

cristinamartinrod replied:
The same

yajo commented on mass_mailing_newsletter_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,32 @@
+            <field name="name">Send Welcome Mail to new subscribers on Newsletter</field>
+            <field name="model_id" ref="mass_mailing.model_mail_mass_mailing_contact"/>
+            <field name="state">email</field>
+            <field name="trigger">on_create</field>
```
This won't take into account if the user is added by a backend odoo user or if he subscribes himself. Is that what we want? :thinking:

pedrobaeza replied:
I don't think we are able to differentiate both.

cristinamartinrod replied:
Now I can not remember any email marketing software that does not send an automated welcome email for new subscribers to a list regardless of the source (import, create, add, forms, ...). It's basic functionality, I suppose that if an Odoo user wants that action to exclude x subscribers they'll have to modify the filter of the action or create a customized one.

yajo replied:
We can differentiate if the mail sending is triggered via the subscription controller instead of from an automated action.

I think current approach can lead to surprises...

yajo commented on mass_mailing_newsletter_welcome_mail/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,3 @@
+This module was written to extend the functionality of the website popup
+newwsletter subscription form to support sending to new subscribers an
```
```suggestion
newsletter subscription form to support sending to new subscribers an
```

cristinamartinrod replied:
Done

yajo commented on mass_mailing_newsletter_welcome_mail/i18n/mass_mailing_welcome_mail.pot:
```diff
@@ -0,0 +1,35 @@
+                    <br/>
+                    <h3>Thanks for signing up four our newsletter!</h3>
+                    <p>
+                        Enjoy a discount code for 20% off your first order
```
You have to update the translations.

cristinamartinrod replied:
Yes, with last change of this template doesn't update the .pot. It's already changed

yajo commented on mass_mailing_newsletter_welcome_mail/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    'depends': [
+        'mass_mailing',
+        'website',
+        'website_crm',
```
Why `website` and `website_crm`? :thinking:

cristinamartinrod replied:
Testing website forms, but it's not necessary that dependency. I remove it

yajo commented on mass_mailing_newsletter_welcome_mail/data/base_automation_data.xml:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+    <data>
```
You don't need `<odoo>` and `<data>`. Just use one.

cubells reviewed (approved):
:+1: 

LGTM

Code review and test, but do the changes taht @Yajo recommend.

yajo committed:
[IMP] mass_mailing_welcome_mail: More logical workflow

- Allow user to customize template.
- Only react when someone subscribes from the website. Not when creating a contact from the backend.
- Proper usage of context values.
- Full instructions.

yajo reviewed (approved):
I fixed all myself:

- Allow user to customize template.
- Only react when someone subscribes from the website. Not when creating a contact from the backend.
- Proper usage of context values.
- Full instructions.

I cannot write tests because of https://github.com/odoo/odoo/pull/30768, so I added it to roadmap.

Tardo commented:
website_mass_mailing can be used to increase database, has no limits in email length... malicious people can create an script to generate records on demand.

I think that 'mass_mailing_newsletter_welcome_mail' can be used to spam an email account... because never checks if the contact is new. But i don't test this.

pedrobaeza commented:
@Tardo interesting thoughts about this. I also think it's redundant, but it was something specifically asked by a customer against our advice. Any way, possible spammers won't control the mail template that is sent, so they wouldn't get their goal of spamming with something specific. Only "noise". And this can't be done several times (only the first one), as Odoo control if an email already belongs to the mailing list.

Tardo commented:
@pedrobaeza yeah.. I had to say "flood" instead of "spam": Perhaps fill the email quota... or (i don't know how really this work) get a ban for the domain from email providers :/

pedrobaeza commented:
OK, then let's continue adding that warning in the README, OK?

yajo commented:
This is a problem in the base addon. I reported it here: https://github.com/odoo/odoo/issues/33234, you can put the link in the README If you want.

pedrobaeza commented:
Can you please put it in the roadmap?
